dell/src/main/java/org/apache/iceberg/dell/ecs/EcsOutputFile.java

```diff
@@ -25,18 +25,24 @@
 import org.apache.iceberg.io.InputFile;
 import org.apache.iceberg.io.OutputFile;
 import org.apache.iceberg.io.PositionOutputStream;
+import org.apache.iceberg.metrics.MetricsContext;
 
 class EcsOutputFile extends BaseEcsFile implements OutputFile {
   public static EcsOutputFile fromLocation(String location, S3Client client) {
-    return new EcsOutputFile(client, new EcsURI(location), new DellProperties());
+    return new EcsOutputFile(client, new EcsURI(location), new DellProperties(), MetricsContext.nullMetrics());
   }
 
   public static EcsOutputFile fromLocation(String location, S3Client client, DellProperties dellProperties) {
-    return new EcsOutputFile(client, new EcsURI(location), dellProperties);
+    return new EcsOutputFile(client, new EcsURI(location), dellProperties, MetricsContext.nullMetrics());
```

Comment:
Should we deprecate this factory method?

Reply:
Yes, that should be in-line with the other FileIO's. I have deprecated both the public static methods.